client/control/components/data/QueryAPIToken.js

```diff
@@ -0,0 +1,38 @@
+import { Component, PropTypes as pt } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { requestAPIToken } from '../../actions/authActions.js';
+
+class QueryAPIToken extends Component {
+  static propTypes = {
+    tokenFetching: pt.bool,
+    token: pt.object,
+    requestAPIToken: pt.func.isRequired,
+  }
+
+  componentWillMount() {
+    this.request();
+  }
+
+  request() {
+    if (!this.props.tokenFetching && !this.props.token) {
```

Comment:
newProps.token would be the thing being requested, so it won't need to trigger a new request. It is possible that the API token is deleted on the server, and would need re-requested, but there isn't anyway for this component to tell if that is the case.

Having a `componentWillReceiveProps` that just calls `request` is probably a good idea though.